homeassistant/core.py

```diff
@@ -1330,7 +1330,7 @@ def register(
         self,
         domain: str,
         service: str,
-        service_func: Callable,
+        service_func: Callable[[ServiceCall], Awaitable | None],
```

Comment:
The awaitable method should have a `None` return type, right?

```suggestion
        service_func: Callable[[ServiceCall], Awaitable[None] | None],
```

Reply:
I have adjusted it, and took the opportunity to make the same change in `async_register_admin_service`